tests/test_locality_NeighborQuery.py

```diff
@@ -375,87 +375,83 @@ def test_exclude_ii(self):
 
         assert ij1 == ij2
 
-    def test_exhaustive_search(self):
+    i = ("i", list(range(10)))
+
+    @pytest.mark.parametrize(*i)
+    def test_exhaustive_search(self, i):
```

Comment:
Seeing a variable `i` like this is a red flag. The convention of names like `i` is to represent a single integer, and usually only defined in the scope of a loop. This `i` is a list of integers, and any code in this class that uses the variable `i` will implicitly shadow the `i` defined here, which isn't great.

Upon further investigation, I see that `i` is only used to enumerate the test's random seeds. I suggest something like:
```suggestion
    @pytest.mark.parametrize("seed", range(1, 10))
    def test_exhaustive_search(self, seed):
```
and then remove `seed = 0` and replace `seed=seed + i` with `seed=seed` below.